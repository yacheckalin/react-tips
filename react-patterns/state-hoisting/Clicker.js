import React from "react";

class Clicker extends React.Component {
  constructor() {
    super();
    this.state = { clicked: 0 };
  }

  handleClick = (e) => {
    this.setState((state) => (this.state.clicked = state.clicked + 1));
  };

  render() {
    return (
      <div>
        <Button
          onClick={this.handleClick}
          name="Clicker"
          className="btn btn-large waves-effect waves-light"
        />
        <p>You clicked : {this.state.clicked} times!</p>
      </div>
    );
  }
}

const Button = ({ onClick, name, className = "btn" }) => (
  <a className={className} onClick={(e) => onClick(e.target)}>
    {name}
  </a>
);

export default Clicker;
