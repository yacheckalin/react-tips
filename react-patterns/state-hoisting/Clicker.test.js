import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";
import Clicker from "./Clicker";

describe("<Clicker />", () => {
  it("renders and displays properly", () => {
    const wrapper = shallow(<Clicker />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
