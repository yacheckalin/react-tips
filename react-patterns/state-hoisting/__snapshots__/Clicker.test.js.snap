// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clicker /> renders and displays properly 1`] = `
<div>
  <Button
    className="btn btn-large waves-effect waves-light"
    name="Clicker"
    onClick={[Function]}
  />
  <p>
    You clicked : 
    0
     times!
  </p>
</div>
`;
